src/docs/sphinx/dev_guide/updating_tpls.rst

```diff
@@ -25,6 +25,108 @@ these dependencies for Axom.
+
+Level 1: Spack
+--------------
+
+Spack is a multi-platform package manager that builds and installs multiple versions
+and configurations of software packages. It has recipes on how to build each package
+with variants on each package to customize them to your needs.  For example, Axom
+has variants for Fortran and MPI, among others.  These recipes handle how to drive
+the individual packages build systems, as well as any packages they depend on.
+Spack also handles system level packages, so you can describe where they are on your
+system instead of building them from scratch.  You will need to describe which compilers
+are available on your system as well.
+
+* Platform specific configuration files live under ``scripts/spack/configs/<platform name>``.
+  There are two files per platform:
+
+   * ``compilers.yaml``: This file contains the compiler specs that describe the location
+     and any other required information about that compiler.  For example, compiler or 
+     linker flags.
+   * ``packages.yaml``: This file describes the system level packages.  For example,
+     where they are located and what version they are. This file is very imporant
+     due to its ability to drastically reduce the amount of packages that Spack builds.
+
+* Axom specific Spack package files live under ``scripts/spack/packages``. These override
+  the package files that live in Spack's repository here ``var/spack/repos/builtin/packages``.
+  We try to minimize these but we have have to alter the existing packages to apply fixes before
+  pushing them up to Spack proper or alterations to the recipes that are Axom specific.
+  This overriding does not happen at the Spack level, but at the next level, Uberenv.
+* `Spack's Github repo <https://github.com/spack/spack>`_
+* `Spack's documentation <https://spack.readthedocs.io/en/latest/>`_
+
+Level 1: Vcpkg
+--------------
+
+Vcpkg is an open-source C++ Library Manager for Windows, Linux, and MacOS by Microsoft.
+Axom only uses it for our Windows TPL builds.
+
+* Project specific package files live under ``develop/scripts/vcpkg_ports``.  There are
+  two different files for each package:
+
+   * ``portfile.cmake``: This file is the recipe on how to build the package. Vcpkg
+     has strict rules about how your project is laid out and you can do the conversion
+     in this file as well.
+   * ``vcpkg.json``: This is the manifest file that describes information about the
+     package.  For example, dependencies, license information, and optional features.
+
+* `Vcpkg's Github repo <https://github.com/microsoft/vcpkg>`_
+* `Vcpkg's documentation <https://github.com/microsoft/vcpkg#table-of-contents>`_
+
+Level 2: Uberenv
+----------------
+
+Uberenv simplifies the use of two level 1 package managers, Spack and Vcpkg.
+We rely on Uberenv for two major points: reducing multiple commands into one
+and adding as much determinism as possible. The basic workflow in Uberenv is
```

Comment:
What is meant by determinism?

Reply:
The same outcome every time.  So if I don't change anything and am on the same platform and using the same compiler, I will get the same set of TPLs built in the same way (versions, options, etc).

Reply:
Now you want me to document my documentation. That is a step too far sir!

(I'll add it in a bit!)